core/src/rpc.rs

```diff
@@ -77,17 +77,31 @@ pub struct JsonRpcRequestProcessor {
 }
 
 impl JsonRpcRequestProcessor {
-    fn bank(&self, commitment: Option<CommitmentConfig>) -> Arc<Bank> {
+    fn bank(&self, commitment: Option<CommitmentConfig>) -> Result<Arc<Bank>> {
         debug!("RPC commitment_config: {:?}", commitment);
         let r_bank_forks = self.bank_forks.read().unwrap();
         if commitment.is_some() && commitment.unwrap().commitment == CommitmentLevel::Recent {
             let bank = r_bank_forks.working_bank();
             debug!("RPC using working_bank: {:?}", bank.slot());
-            bank
+            Ok(bank)
         } else {
-            let slot = r_bank_forks.root();
-            debug!("RPC using block: {:?}", slot);
-            r_bank_forks.get(slot).cloned().unwrap()
+            let cluster_root = self
+                .block_commitment_cache
+                .read()
+                .unwrap()
+                .largest_confirmed_root();
+            debug!("RPC using block: {:?}", cluster_root);
+            r_bank_forks
+                .get(cluster_root)
+                .cloned()
+                .ok_or_else(|| Error {
+                    code: ErrorCode::ServerError(-32000),
```

Comment:
What's `-32000` from?  Is there a constant that can be used?

Reply:
That's the first available server error in the jsonrpc spec. Good call. I'll use a single constant for now, but potentially something more elaborate when we implement more jsonrpc error objects down the road.